modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -1653,6 +1696,41 @@ bool cv::gimpl::GStreamingExecutor::pull(cv::GOptRunArgsP &&outs)
     return true;
 }
 
+std::tuple<bool, cv::util::variant<cv::GRunArgs, cv::GOptRunArgs>> cv::gimpl::GStreamingExecutor::pull()
+{
+    using RunArgs = cv::util::variant<cv::GRunArgs, cv::GOptRunArgs>;
+    bool is_over = false;
+
+    cv::gimpl::GModel::ConstGraph cgr(*m_orig_graph);
+    auto meta = cgr.metadata().get<cv::gimpl::Protocol>().out_nhs;
+    cv::GTypesInfo info;
+    info.reserve(meta.size());
+
+    ade::util::transform(meta, std::back_inserter(info), [&cgr](const ade::NodeHandle& nh) {
+        const auto& data = cgr.metadata(nh).get<cv::gimpl::Data>();
+        return cv::GTypeInfo{data.shape, data.kind, data.ctor};
+    });
```

Comment:
I believe you shouldn't do this on every `pull()`, this can be done once at the executor construction time.

No need to fix it right now as it doesn't affect the main (non-Python) path but please file a task to do it somewhere later.

Reply:
Fixed.